server/middleware/token.js

```diff
@@ -68,10 +68,16 @@ function escapeRegExp(str) {
 
 function token(options) {
   options = options || {};
+
+  var registry = options.registry;
+  if(!registry && options.app && options.app.registry) {
+    registry = options.app.registry;
+  }
+  
   var TokenModel = options.model || loopback.AccessToken;
```

Comment:
This is wrong, you can't use `loopback.AccessToken` as the default, you need to use the app-local model instead.

The solution is to move the code resolving TokenModel to the middleware, because then you have access to the loopback app object.

```js
function token(options) {
  var TokenModel;
  // etc.
  return function(req, res, next) {
    if (TokenModel === undefined) {
      var app = req.app;
      var registry = app.registry;
      TokenModel = options.model || registry.AccessToken;
      if (typeof TokenModel === 'string') {
        // Make it possible to configure the model in middleware.json
        TokenModel = registry.getModel(TokenModel);
      }
    }
    // etc.
  };
}
```